dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java

```diff
@@ -201,6 +217,29 @@ public void patch(Context context, HttpServletRequest request, String apiCategor
                 throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + uuid + " not found");
             }
             List<Operation> operations = patch.getOperations();
+
+            /**
+             * Although users can update their own password, other
+             * patch operations are for administrators only. Throwing
+             * the error from here.
+             */
+            boolean isAdmin;
+            try {
+                isAdmin = authorizeService.isAdmin(context);
+            } catch (SQLException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+            for (Operation op: operations) {
+                if (op.getPath().contentEquals("/canLogin")
+                        || op.getPath().contentEquals("/certificate")
+                        || op.getPath().contentEquals("/netid")
+                        || op.getPath().contentEquals("/email")) {
+                    if (!isAdmin) {
+                        throw new AuthorizeException(op.getPath() + " requires administrator.");
+                    }
+                }
+            }
```

Comment:
Hi @mspalti,

Ideally this code should move to a Permission evaluator so that this method can [just focus on doing the patch operation](https://en.wikipedia.org/wiki/Single_responsibility_principle). In addition, we might encounter similar scenario's for other DSpace objects.

This might seem like a challenge, but I think it's doable:

1. Change the `hasPermission(#uuid, 'EPERSON', 'WRITE')` at the top of this method to `hasPermission(#uuid, 'EPERSON', #patch)`.

1. In the `DSpaceObjectRestPermissionEvaluator` super class, implement the `boolean hasPermission(Authentication authentication, Serializable targetId, String targetType, Object permission)` method as follows: Check if `permission instanceof Patch`. If that is the case, call a new method `hasPatchPermission(authentication, targetId, targetType, (Patch) permission)`. Otherwise convert the permission to a `DSpaceRestPermission` (`DSpaceRestPermission restPermission = DSpaceRestPermission.convert(permission);`) and call a new abstract method `hasDSpacePermission(authentication, targetId, targetType, restPermission)`.

1. Add a default implementation for `hasPatchPermission` that just does `return hasPermission(authentication, targetId, targetType, "WRITE");`

1. Implement the `hasDSpacePermission` in all subclasses by just renaming the currently present `hasPermission` method and changing the type of the last parameter.

1. In the `EPersonRestPermissionEvaluatorPlugin`, override the `hasPatchPermission` method. In that implementation, first check that `hasPermission(authentication, targetId, targetType, "WRITE")` is true. If that is the case, then execute the custom admin checks you wrote here.

Then you should be able to remove these admin checks from the `EPersonRestRepository` class.

Reply:
I agree with @tomdesair proposal. If it looks to complex I'm also fine with log it as a necessary improvement and deal with it in a followup PR

Reply:
I will work on @tomdesair suggestion.

Reply:
Hi @tomdesair 

I finally got to work on your proposal and it appears to be successful. The check for patch permission on the eperson is now handled by the `EPersonRestPermissionEvaluatorPlugin`. All tests are passing...